ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -4357,7 +4357,7 @@ IoGetDeviceProperty(IN PDEVICE_OBJECT DeviceObject,
             break;
 
         case DevicePropertyRemovalPolicy:
-            PIP_RETURN_DATA(sizeof(UCHAR), &DeviceNode->RemovalPolicy);
+            PIP_RETURN_DATA(sizeof(DEVICE_REMOVAL_POLICY), &DeviceNode->RemovalPolicy);
```

Comment:
You may be right that the caller perhaps expects 4 bytes returned, but `DEVICE_NODE::RemovalPolicy` is a UCHAR, and returning the data beyond that will include the next field (`HardwareRemovalPolicy`) as well as two bytes of padding.
So if this needs to be two bytes, you probably need to create a local variable.